src/signed-data-store/signed-data-store.ts

```diff
@@ -0,0 +1,130 @@
+import { clearInterval } from 'node:timers';
+import { BigNumber, ethers } from 'ethers';
+import { logger } from '../logger';
+import type { LocalSignedData, SignedData, AirnodeAddress, TemplateId } from '../types';
+
+// A simple in-memory data store implementation - the interface allows for swapping in a remote key/value store
+
+let signedApiStore: Record<AirnodeAddress, Record<TemplateId, LocalSignedData>> = {};
+let pruner: NodeJS.Timeout | undefined;
+
+export const verifySignedData = ({ airnode, templateId, timestamp, signature, encodedValue }: SignedData) => {
+  // 'encodedValue' is: ethers.utils.defaultAbiCoder.encode(['int256'], [beaconValue]);
+
+  const message = ethers.utils.arrayify(
+    ethers.utils.solidityKeccak256(['bytes32', 'uint256', 'bytes'], [templateId, timestamp, encodedValue])
+  );
+
+  const signerAddr = ethers.utils.verifyMessage(message, signature);
+  return signerAddr !== airnode;
+};
+
+const checkTimestamp = ({ timestamp, airnode, encodedValue, templateId }: SignedData) => {
+  if (Number.parseInt(timestamp, 10) * 1000 > Date.now() + 60 * 60 * 1000) {
+    logger.warn(
+      `Refusing to store sample as timestamp is more than one hour in the future: (Airnode ${airnode}) (Template ID ${templateId}) (Received timestamp ${new Date(
+        Number.parseInt(timestamp, 10) * 1000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+    return false;
+  }
+
+  if (Number.parseInt(timestamp, 10) * 1000 > Date.now()) {
+    logger.warn(
+      `Sample is in the future, but by less than an hour, therefore storing anyway: (Airnode ${airnode}) (Template ID ${templateId}) (Received timestamp ${new Date(
+        Number.parseInt(timestamp, 10) * 1000
+      ).toLocaleDateString()} vs now ${new Date().toLocaleDateString()}), ${
+        BigNumber.from(encodedValue).div(10e10).toNumber() / 10e8
+      }`
+    );
+  }
+
+  return true;
+};
+
+export const checkSignedDataIntegrity = (signedData: SignedData) => {
```

Comment:
Some of these functions are called `verifyXyz` and some `checkXyz`. It's better to stick to one. I personally like "verify" more.